### PR TITLE
refactoring hosts

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

File: demos/asp-net-core/Divergent.CompositionGateway/Program.cs [hunk 1]
```diff
@@ -5,17 +5,14 @@ namespace Divergent.CompositionGateway
 {
     public class Program
     {
-        public static void Main(string[] args)
-        {
-            var host = new WebHostBuilder()
+        public static void Main(string[] args) =>
```
Comment (mauroservienti) [inline demos/asp-net-core/Divergent.CompositionGateway/Program.cs]: This was created as the original VS/new project template, I don't think we should change it, it'll only raise time wasting questions

File: demos/asp-net-core/Divergent.CompositionGateway/Program.cs [hunk 2]
```diff
@@ -5,17 +5,14 @@ namespace Divergent.CompositionGateway
...
                 .UseContentRoot(Directory.GetCurrentDirectory())
                 .UseIISIntegration()
                 .UseStartup<Startup>()
                 .UseApplicationInsights()
-                .Build();
-
-            host.Run();
-        }
+                .Build()
+                .Run();
```
Comment (mauroservienti) [inline demos/asp-net-core/Divergent.CompositionGateway/Program.cs]: This was created as the original VS/new project template, I don't think we should change it, it'll only raise time wasting questions

File: demos/asp-net-core/Divergent.CompositionGateway/Startup.cs [hunk 3]
```diff
@@ -8,15 +8,12 @@ namespace Divergent.CompositionGateway
...
     {
-        public void ConfigureServices(IServiceCollection services)
-        {
-            services.AddRouting();
-            services.AddViewModelComposition();
-        }
+        public void ConfigureServices(IServiceCollection services) =>
+            services
+                .AddRouting()
+                .AddViewModelComposition();
```
Comment (mauroservienti) [inline demos/asp-net-core/Divergent.CompositionGateway/Startup.cs]: This was created as the original VS/new project template, I don't think we should change it, it'll only raise time wasting questions

File: demos/asp-net-core/Divergent.CompositionGateway/Startup.cs [hunk 4]
```diff
@@ -8,15 +8,12 @@ namespace Divergent.CompositionGateway
...
 
-        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
-        {
```
Comment (mauroservienti) [inline demos/asp-net-core/Divergent.CompositionGateway/Startup.cs]: This was created as the original VS/new project template, I don't think we should change it, it'll only raise time wasting questions

File: demos/asp-net-core/Divergent.Sales.API.Host/ServiceHost.cs [hunk 5]
```diff
@@ -1,27 +0,0 @@
...
-    class ServiceHost
-    {
-        IDisposable webApp;
-
-        public void Start()
-        {
-            var baseAddress = ConfigurationManager.AppSettings["baseAddress"];
-
-            webApp = WebApp.Start<Startup>(url: baseAddress);
-            Console.WriteLine($"Divergent.Sales.API.Host listening on {baseAddress}");
```
Comment (mauroservienti) [inline demos/asp-net-core/Divergent.Sales.API.Host/ServiceHost.cs]: it's handy to have the `baseAddress` printed on the console, especially if something goes wrong or you want to demo live something else.

File: demos/asp-net-core/Divergent.Sales.API.Host/Controllers/OrdersController.cs [hunk 6]
```diff
@@ -12,40 +12,36 @@ public class OrdersController : ApiController
         [HttpGet, Route("{orderNumber}")]
         public dynamic Get(int orderNumber)
```
Comment (mauroservienti) [inline demos/asp-net-core/Divergent.Sales.API.Host/Controllers/OrdersController.cs]: if this PR is meant to refactor hosts this file should not be changed.

Comment (adamralph) [reply demos/asp-net-core/Divergent.Sales.API.Host/Controllers/OrdersController.cs]: "Host" as in the "API host" 😉

File: demos/asp-net-core/Divergent.Sales.API.Host/ServiceHost.cs (hunk 5 above)
Comment (adamralph) [reply demos/asp-net-core/Divergent.Sales.API.Host/ServiceHost.cs]: OK, I'll put that back in.

**Comment by adamralph** [comment]
@mauroservienti I pushed a fixup commit which reverts the gateway files to the VS templates, and restores the console output for the API hosts.

**Pull request merged**